File: class12/python/GQS1201-PrimeiroTeste.py
#!/usr/bin/env python3
"""
GQS1201 - Exemplo "Hello Test" de Automação de Testes.
Demonstra o padrão AAA (Arrange-Act-Assert).
Uso: python3 GQS1201-PrimeiroTeste.py

Para executar os testes:
  pytest GQS1201-PrimeiroTeste.py -v
"""


def somar(a: int, b: int) -> int:
    """Função simples a ser testada."""
    return a + b


def test_soma_deve_retornar_cinco():
    """Exemplo de teste com padrão AAA."""
    # Arrange
    a, b = 2, 3
    # Act
    resultado = somar(a, b)
    # Assert
    assert resultado == 5, f"Esperado 5, obtido {resultado}"


def test_soma_com_zero():
    assert somar(0, 5) == 5


def test_soma_com_negativos():
    assert somar(-1, 1) == 0


def test_soma_valores_grandes():
    assert somar(1000000, 2000000) == 3000000


def test_soma_duas_vezes():
    """Demonstra independência de testes."""
    assert somar(3, 4) == 7
    assert somar(10, 20) == 30


if __name__ == "__main__":
    print("GQS1201 - Executando testes manualmente:")
    print(f"  somar(2, 3) = {somar(2, 3)}")
    print(f"  somar(0, 5) = {somar(0, 5)}")
    print(f"  somar(-1, 1) = {somar(-1, 1)}")
    print("\nExecute com 'pytest GQS1201-PrimeiroTeste.py -v'")

File: class12/java/GQS1201-PrimeiroTeste.java
/**
 * GQS1201 - Exemplo "Hello Test" de Automação de Testes.
 * Demonstra o padrão AAA (Arrange-Act-Assert) sem depender de um framework
 * de testes; os métodos testeXxx() usam a instrução assert do Java.
 *
 * Compilação e execução:
 *   javac GQS1201-PrimeiroTeste.java
 *   java -ea -cp . GQS1201_PrimeiroTeste
 *
 * (a flag -ea habilita as instruções assert; sem ela os testes não falham)
 *
 * Para um projeto real, este mesmo cenário seria escrito com JUnit 5,
 * como visto na Aula 13.
 */
class GQS1201_PrimeiroTeste {

    /** Função simples a ser testada. */
    static int somar(int a, int b) {
        return a + b;
    }

    /** Exemplo de teste com padrão AAA. */
    static void testeSomaDeveRetornarCinco() {
        // Arrange
        int a = 2, b = 3;
        // Act
        int resultado = somar(a, b);
        // Assert
        assert resultado == 5 : "Esperado 5, obtido " + resultado;
    }

    static void testeSomaComZero() {
        assert somar(0, 5) == 5;
    }

    static void testeSomaComNegativos() {
        assert somar(-1, 1) == 0;
    }

    static void testeSomaValoresGrandes() {
        assert somar(1000000, 2000000) == 3000000;
    }

    /** Demonstra independência de testes. */
    static void testeSomaDuasVezes() {
        assert somar(3, 4) == 7;
        assert somar(10, 20) == 30;
    }

    public static void main(String[] args) {
        System.out.println("GQS1201 - Executando testes manualmente:");
        System.out.printf("  somar(2, 3) = %d%n", somar(2, 3));
        System.out.printf("  somar(0, 5) = %d%n", somar(0, 5));
        System.out.printf("  somar(-1, 1) = %d%n", somar(-1, 1));

        testeSomaDeveRetornarCinco();
        testeSomaComZero();
        testeSomaComNegativos();
        testeSomaValoresGrandes();
        testeSomaDuasVezes();

        System.out.println("\nTodos os testes executados. Rode com 'java -ea' para validar os asserts.");
    }
}
